src/vs/workbench/services/workspace/node/workspaceMigrationService.ts

```diff
@@ -38,32 +38,42 @@ export class WorkspaceMigrationService implements IWorkspaceMigrationService {
 
 	migrate(toWorkspaceId: IWorkspaceIdentifier): TPromise<void> {
 		this.migrateStorage(toWorkspaceId);
+
 		return this.migrateConfiguration(toWorkspaceId);
 	}
 
 	migrateStorage(toWorkspaceId: IWorkspaceIdentifier): void {
+
 		// The shutdown sequence could have been stopped due to a veto. Make sure to
 		// always dispose the shutdown listener if we are called again in the same session.
 		if (this.shutdownListener) {
 			this.shutdownListener.dispose();
 			this.shutdownListener = void 0;
 		}
 
+		// Since many components write to storage only on shutdown, we register a shutdown listener
+		// very late to be called as the last one.
 		this.shutdownListener = once(this.lifecycleService.onShutdown)(() => {
+
 			// TODO@Ben revisit this when we move away from local storage to a file based approach
 			const storageImpl = this.storageService as StorageService;
 			migrateStorageToMultiRootWorkspace(storageImpl.storageId, toWorkspaceId, storageImpl.workspaceStorage);
 		});
 	}
 
 	private migrateConfiguration(toWorkspaceId: IWorkspaceIdentifier): TPromise<void> {
+		if (!this.contextService.hasFolderWorkspace()) {
```

Comment:
@sandy081 fyi I changed this code to only run when we are in a folder workspace. The reason for that is that with the format changes I will run the migration not only when going from 1 folder to a workspace but also when saving a workspace to another location on disk.

Reply:
@bpasero Code looks like migration does not run only when we are in folder workspace?